From 71f037bfa2a72c49075869ddc38107300f7caf7a Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Sun, 5 May 2019 17:36:57 +0300
Subject: [PATCH 1121/1200] dts: marvell: fix efuse entries and OTP base
 address

Change the CP LD efuse names to avoid naming collisions when
more than one CP is connected to AP.
Replace OTP address with offset - the base OTP address should
be calculated and added by the efuse driver
Update A38x DTSI for reflecting OTP address changes

Change-Id: I9018d0c2dd95fc0f2d3cc0e017a492aeb59bb998
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/8592
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
---
 arch/arm/dts/armada-38x.dtsi   | 2 +-
 arch/arm/dts/armada-ap80x.dtsi | 6 +++---
 arch/arm/dts/armada-cp110.dtsi | 8 ++++----
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/arch/arm/dts/armada-38x.dtsi b/arch/arm/dts/armada-38x.dtsi
index de0b11c5b1..cd6c405a63 100644
--- a/arch/arm/dts/armada-38x.dtsi
+++ b/arch/arm/dts/armada-38x.dtsi
@@ -598,7 +598,7 @@
 			efuse: efuse@f9000 {
 				compatible = "marvell,mvebu-fuse-hd";
 				reg = <0xe4008 0x4>;
-				otp-mem = <0xf60f9000>;
+				otp-mem = <0xf9000>;
 				rows-count = <64>;
 				status = "disabled";
 			};
diff --git a/arch/arm/dts/armada-ap80x.dtsi b/arch/arm/dts/armada-ap80x.dtsi
index 444261ecbd..348e965bd5 100644
--- a/arch/arm/dts/armada-ap80x.dtsi
+++ b/arch/arm/dts/armada-ap80x.dtsi
@@ -110,21 +110,21 @@
 			ap_ld_efuse0: efuse-0@6F8F00 {
 				compatible = "marvell,mvebu-fuse-ld-user";
 				reg = <0x6F8008 0x4>;
-				otp-mem = <0xF06F8F00>;
+				otp-mem = <0x6F8F00>;
 				status = "disabled";
 			};
 
 			ap_ld_efuse1: efuse-1@6F8F00 {
 				compatible = "marvell,mvebu-fuse-ld-prop";
 				reg = <0x6F8008 0x4>;
-				otp-mem = <0xF06F8F00>;
+				otp-mem = <0x6F8F00>;
 				status = "disabled";
 			};
 
 			ap_hd_efuse0: efuse@6F9000  {
 				compatible = "marvell,mvebu-fuse-hd";
 				reg = <0x6F8008 0x4>;
-				otp-mem = <0xF06F9000>;
+				otp-mem = <0x6F9000>;
 				rows-count = <64>;
 				status = "disabled";
 			};
diff --git a/arch/arm/dts/armada-cp110.dtsi b/arch/arm/dts/armada-cp110.dtsi
index 481165add2..7668f06e8b 100644
--- a/arch/arm/dts/armada-cp110.dtsi
+++ b/arch/arm/dts/armada-cp110.dtsi
@@ -55,17 +55,17 @@
 				status = "okay";
 			};
 
-			CP110_LABEL(ld_efuse0): efuse0@400F00 {
+			CP110_LABEL(ld_efuse0): CP110_LABEL(efuse0)@400F00 {
 				compatible = "marvell,mvebu-fuse-ld-user";
 				reg = <0x400008 0x4>;
-				otp-mem = <0xF2400F00>;
+				otp-mem = <0x400F00>;
 				status = "disabled";
 			};
 
-			CP110_LABEL(ld_efuse1): efuse1@400F00 {
+			CP110_LABEL(ld_efuse1): CP110_LABEL(efuse1)@400F00 {
 				compatible = "marvell,mvebu-fuse-ld-prop";
 				reg = <0x400008 0x4>;
-				otp-mem = <0xF2400F00>;
+				otp-mem = <0x400F00>;
 				status = "disabled";
 			};
 
-- 
2.22.0

